Cleanup GIC-v2/GIC-v3 code

marcrittinghaus opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [x] Updated relevant documentation.


### Base target

 - Architecture(s): [`arm64`]
 - Platform(s): [N/A]
 - Application(s): [N/A]


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

This PR bundles some cleanups for the GIC-v2 implementation that improves documentation, adds additional assertations and renames the `gic_`-prefix to `gicv2_` to be inline with the GIC-v3 implementation. The PR also fixes a warning about a wrong format specifier used for `__lcpuid` in GIC-v2 and GIC-v3.

marcrittinghaus committed:
plat/drivers/gic-v2: Improve comments

This is a cleanup commit to improve the documentation for
the GICv2 functions

Signed-off-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

marcrittinghaus committed:
plat/drivers/gic-v2: Add asserts

Some of the parameters allow larger value ranges than accepted by the
GIC. We therefore add additional assertations to make sure values are
valid. We also replace the crash in `gic_set_irq_target` as specifying a
wrong IRQ number can be considered a bug instead of a runtime error.

Signed-off-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

marcrittinghaus committed:
plat/drivers/gic-v2: Rename functions to gicv2_

To be inline with the GIC-v3 implementation, we change the
prefix of all GIC-v2 function from `gic_` to `gicv2_`.

Signed-off-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

marcrittinghaus committed:
plat/drivers/gic: Fix format specifiers in dbg msg

The __lcpuid type changed from __u32 to __u64. This commit fixes the
warning in the GIC-v2 and GIC-v3 implementation.

Signed-off-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

rene commented on plat/drivers/gic/gic-v2.c:
```diff
@@ -271,10 +303,10 @@ static void gic_enable_dist(void)
 }
 
-/* disable distributor */
+/** Enable distributor */
```
Should be "Disable"

rene commented on plat/drivers/gic/gic-v2.c:
```diff
@@ -154,7 +171,7 @@ static void gic_sgi_gen(uint32_t sgintid, enum sgi_filter targetfilter,
 	UK_ASSERT(sgintid <= GICD_SGI_MAX_INITID);
 
-	/* Set SGI tagetfileter field */
+	/* Set SGI tagetfilter field */
```
tagetfilter -> ta**r**getfilter

rene commented:
@marcrittinghaus, in commit 28e0ad6fc536c951ff314cd3ab0d4933b459bdc8 I would change "Fix comments" to "Improve comments"... by reading for the first time it gave me the impression that comments were all wrong....

marcrittinghaus commented:
> @marcrittinghaus, in commit [28e0ad6](https://github.com/unikraft/unikraft/commit/28e0ad6fc536c951ff314cd3ab0d4933b459bdc8) I would change "Fix comments" to "Improve comments"... by reading for the first time it gave me the impression that comments were all wrong....

Hey @rene,
thank you very much for looking over this. I fixed the typos and changed the commit message 😃

rene commented:
Thanks @marcrittinghaus 

Reviewed-by: Renê de Souza Pinto <rene@renesp.com.br>
Approved-by: Renê de Souza Pinto <rene@renesp.com.br>
